src/Service/Statistics/FilteredBackAnalyticsProvider.php

```diff
@@ -0,0 +1,124 @@
+    public $hasCountRefused;
+    public $hasCountArchived;
+    public $territory;
+
+    public function __construct(
+        private SignalementRepository $signalementRepository,
+        private TerritoryRepository $territoryRepository,
+        private MonthStatisticProvider $monthStatisticProvider,
+        private PartenaireStatisticProvider $partenaireStatisticProvider,
+        private SituationStatisticProvider $situationStatisticProvider,
+        private CriticiteStatisticProvider $criticiteStatisticProvider,
+        private StatusStatisticProvider $statusStatisticProvider,
+        private CriticitePercentStatisticProvider $criticitePercentStatisticProvider,
+        private VisiteStatisticProvider $visiteStatisticProvider,
+        private MotifClotureStatisticProvider $motifClotureStatisticProvider,
+        ) {
+    }
+
+    public function initFilters(Request $request, ?Territory $territory)
+    {
+        $this->communes = json_decode($request->get('communes'));
+        $this->statut = $request->get('statut');
+        $strEtiquettes = $request->get('etiquettes');
+        $this->etiquettes = array_map(fn ($value): int => $value * 1, json_decode($strEtiquettes));
+        $this->type = $request->get('type');
+        $dateStart = $request->get('dateStart');
+        $this->dateStart = new DateTime($dateStart);
+        $dateEnd = $request->get('dateEnd');
+        $this->dateEnd = new DateTime($dateEnd);
+        $this->hasCountRefused = '1' == $request->get('countRefused');
+        $this->hasCountArchived = '1' == $request->get('countArchived');
+        $this->territory = $territory;
+    }
+
+    public function getData(): array
+    {
+        $buffer = [];
+        $buffer['count_signalement_filtered'] = $this->getCountSignalementData();
+        $buffer['average_criticite_filtered'] = round($this->getAverageCriticiteData() * 10) / 10;
+        $buffer['count_signalement_per_month'] = $this->getCountSignalementPerMonth();
+        $buffer['count_signalement_per_partenaire'] = $this->getCountSignalementPerPartenaire();
+        $buffer['count_signalement_per_situation'] = $this->getCountSignalementPerSituation();
+        $buffer['count_signalement_per_criticite'] = $this->getCountSignalementPerCriticite();
+        $buffer['count_signalement_per_statut'] = $this->getCountSignalementPerStatut();
+        $buffer['count_signalement_per_criticite_percent'] = $this->getCountSignalementPerCriticitePercent();
+        $buffer['count_signalement_per_visite'] = $this->getCountSignalementPerVisite();
+        $buffer['count_signalement_per_motif_cloture'] = $this->getCountSignalementPerMotifCloture();
+
+        return $buffer;
+    }
+
+    private function getCountSignalementData(): int
+    {
+        return $this->signalementRepository->countFiltered($this);
+    }
+
+    private function getAverageCriticiteData(): float
+    {
+        $buffer = $this->signalementRepository->getAverageCriticiteFiltered($this);
```

Comment:
Mettre en paramètre l'objet courant `$this` rejoint le fait qu'on devrait passer par un objet dédié dont c'est le rôle, pour éviter de passer tous le provider . Ce commentaire s'applique à toutes les méthodes privées qui font appels aux classes repository